di_website/events/models.py

```diff
@@ -0,0 +1,88 @@
+from django.db import models
+from wagtail.admin.edit_handlers import (FieldPanel, InlinePanel,
+                                         MultiFieldPanel, StreamFieldPanel)
+from wagtail.core.fields import StreamField
+from wagtail.core.models import Page
+
+from di_website.common.base import OtherPage, StandardPage, get_paginator_range
+from di_website.common.blocks import BaseStreamBlock
+from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
+from django.contrib.contenttypes.models import ContentType
+
+from taggit.models import Tag, TaggedItemBase
+
+""" Content of each Event that makes up events list """
+
+
+class EventPage(StandardPage):
+
+    template = "events/events_page.html"
+
+    start_time = models.DateTimeField()
+    end_time = models.DateTimeField()
+    location = models.CharField(
+        max_length=100, help_text='Physical location of event')
+
+    body = StreamField(BaseStreamBlock(), verbose_name="Page Body", blank=True)
+    other_pages_heading = models.CharField(
+        blank=True, max_length=255, verbose_name='Heading', default='Other pages in this section')
+
+    content_panels = StandardPage.content_panels + [
+        MultiFieldPanel([
+            FieldPanel('start_time'),
+            FieldPanel('end_time'),
+            FieldPanel('location')
+        ], heading='Event Details'),
+
+        StreamFieldPanel('body'),
+        MultiFieldPanel([
+            FieldPanel('other_pages_heading'),
+            InlinePanel('other_pages', label='Other Pages')
```

Comment:
Try to limit this to the expected pages, which are Blogs, Events or Publications.

Reply:
Its now limited to blog articles and event pages; I have not seen publication models so that is pending